library/core/src/num/uint_macros.rs

```diff
@@ -1216,8 +1215,7 @@ macro_rules! uint_impl {
                       without modifying the original"]
         #[inline]
         pub const fn checked_ilog2(self) -> Option<u32> {
-            // FIXME: Simply use `NonZero::new` once it is actually generic.
-            if let Some(x) = <$NonZeroT>::new(self) {
+            if let Some(x) = NonZero::new(self) {
```

Comment:
Maybe change these to `match` rather than `if let`, to be consistent with the new `isqrt`? Or even `.map(NonZero::<Self>::ilog10)`.

Reply:
`Option::map` is not `const` (not even unstably). `match` does look a bit nicer though.